aerleon/lib/yaml.py

```diff
@@ -330,7 +334,7 @@ def _LoadIncludeFile(base_dir, inc_filename):
     """Open an include file."""
 
     with open(pathlib.Path(base_dir).joinpath(inc_filename), 'r') as include_file:
-        return include_file
+        return include_file.read()
```

Comment:
Do you really want to load this into memory?  

You could just move `with open()` to the one spot where it is called? 

```patch
-                include_file = _LoadIncludeFile(base_dir, inc_filename)
-                include_data = yaml.load(
-                    include_file, Loader=SpanSafeYamlLoader(filename=inc_filename)
-                )
+                with open(pathlib.Path(base_dir).joinpath(inc_filename), 'r') as include_file:
+                    include_data = yaml.load(
+                        include_file, Loader=SpanSafeYamlLoader(filename=inc_filename)
+                    )
```

Reply:
Yeah, `yaml` loader would stream through the file nicely so we are balooning our process memory a bit by loading in the whole file. In my experience policy files - especially includes - do not seem to be so large that this will be noticeable to the user (unlike network/service definitions which can be truly large).

`_LoadIncludeFile` exists purely for unit test mocks, and those tests really could just mock `open` (as do many similar tests). So my suggestion would be to just do away with `_LoadIncludeFile` completely and focus on mocking `open`.

Reply:
https://github.com/aerleon/aerleon/issues/209